firestore/query.go

```diff
@@ -569,8 +579,48 @@ func trunc32(i int) int32 {
 	return int32(i)
 }
 
+// Get returns an array of query's resulting documents.
+func (q Query) Get(ctx context.Context) ([]*DocumentSnapshot, error) {
+	limitedToLast := q.limitToLast
+
+	if q.limitToLast {
+		// Flip order statements before posting a request.
+		for i := range q.orders {
+			if q.orders[i].dir == Asc {
+				q.orders[i].dir = Desc
+			} else {
+				q.orders[i].dir = Asc
+			}
+		}
+		// Swap cursors.
+		q.startVals, q.endVals = q.endVals, q.startVals
+		q.startDoc, q.endDoc = q.endDoc, q.startDoc
+		q.startBefore, q.endBefore = q.endBefore, q.startBefore
+
+		q.limitToLast = false
+	}
+	docs, err := q.Documents(ctx).GetAll()
+	if err != nil {
+		return nil, err
+	}
+	if limitedToLast {
+		// Flip docs order before return.
+		for i, j := 0, len(docs)-1; i < j; {
+			docs[i], docs[j] = docs[j], docs[i]
```

Comment:
Before the query was executed you already flipped the direction. Does this need to be flipped again?

Reply:
IIUC Ben's and Sebastian's explanations, it works like this:

For example, we have such a docs:
doc1
doc2
doc3
doc4

We're setting `limitToLast=2`. Before posting a backend request, we're flipping the orders and turning `limitToLast` into `limit`, so we're reading two docs:
**doc4**
**doc3**
doc2
doc1

And we're flipping the results again before returning them, so user will get them like we never flipped the orders:
doc3
doc4

Reply:
That makes sense, thanks for the explanation.